core/controllers/retrieve.py

```diff
@@ -0,0 +1,33 @@
+from io import BytesIO
+
+import flask
+from flask import abort, request
+from sqlalchemy import select
+
+from core.const import EXCEL_MIMETYPE
+from core.db import db
+from core.db.entities import Submission
+
+
+def retrieve():
+    """Handle the download request and return the originally submitted spreadsheet.
+
+    Select file by:
+    - UUID
+
+    :return: Flask response object containing the requested spreadsheet.
+    """
+    submission_UUID = request.args.get("submission_UUID")
+
+    query = select(Submission.submission_file, Submission.reporting_round).where(Submission.id == submission_UUID)
+
+    try:
+        file_bytes, reporting_round = db.session.execute(query).one()
+    except Exception:
+        return abort(404, "Could not find a file that matches this UUID")
+
+    return flask.send_file(
+        BytesIO(file_bytes), mimetype=EXCEL_MIMETYPE, as_attachment=True, download_name=f"{submission_UUID}.xlsx"
+    ), {
+        "reporting_round": reporting_round
```

Comment:
Does this work? You can only send one request - this looks like two?

Reply:
This works, the reporting round ends in the response headers, we will need to also return fund_type, because that will be expected by the ingest function as well

Reply:
Nice 👍